src/components/NewRuntimeModal.js

```diff
@@ -570,9 +571,12 @@ export const NewRuntimeModal = withModalDrawer({ width: 675 })(class NewRuntimeM
 
     const isCustomImage = selectedLeoImage === CUSTOM_MODE
 
+    const minRequiredMemory = sparkMode ? 7.5 : 3.75
+    const validMachineTypes = _.filter(({ memory }) => memory >= minRequiredMemory, machineTypes)
+    const mainMachineType = _.find({ name: masterMachineType }, validMachineTypes)?.name || (sparkMode ? defaultDataprocMachineType : defaultGceMachineType)
```

Comment:
It is necessary to define `mainMachineType` because in the case of no runtimes existing, we need to distinguish between an initial state (when a user has just opened the modal) vs. after a user choosing a machine type.